### PR TITLE
project-5-refactorPullRequest

### PR DESCRIPTION
**Pull request by IvonneBenitesRodriguez**
**About this project:**

This project is about refactoring our Ruby code following the main **principles of Object Oriented Design**.
We have respected and followed the principle of **Single Responsibility Principle**.
Finally, we have also applied the **best practices for Ruby and followed its style Guide**.

**For it :** <br/>
- We have created a featured branch named project-5-refactor
- We have updated our files named : app.rb, main.rb, teacher.rb, student.rb
- We have added a folder named : modules and stored insided files (create_entities.rb, list_entities.rb, user_interface.rb)
  We consider that adding this folder named modules improve our code organization, promotes code reusability  and 
  helps us to be align with the Object Oriented Design principles.
- We have also refactored our files, for example in our app.rb file , we have placed some of its methods into create_entities.rb file.
- We have removed unused methods .
- We have passed Ruby linters.

**Goal**<br/>
The goal is to understand how to implement the principles of Object Oriented Design and feel comfortable about it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -31,16 +31,6 @@ def add_rentals(book, date)
...
-  end
-
-  private
-
-  def of_age?
-    @age >= 18
-  end
 end
 
 person = Person.new('maximilianus', 22)
```
Comment (Shedrack-Sunday) [inline person.rb]: - [x] Kindly remove the view of the person class which was to test the functionality of a previous milestone. 👏 

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @IvonneBenitesRodriguez @gilberthappi      👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the ruby class features and following the KISS and YANGIS ✔️
- Kudus making the requirement asked      ✔️  

# - Required Changes  🌵  

- [ ]  Kindly take note of the comment i made on a piece of code which should not be in this milestone 
 
# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 1 above)
Comment (IvonneBenitesRodriguez) [reply person.rb]: Dear @Shedrack-Sunday,
thank you for your suggestions, and kind comments,
we appreciate it. We have modified it as you requested, 
thanks again, 
Best wishes

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @IvonneBenitesRodriguez  @gilberthappi      👍 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby class  features  ✔️  
- [x] Kudus on meeting the requirements as pointed the last time out   👏 


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**